tautology in `fjson_object_iter_equal`

d4b3a2def5cb47192212b4682f64325a1c867749 introduced a tautology in `fjson_object_iter_equal`

```
			if ( (iter1->curr_idx == iter2->curr_idx) &&
			     (iter2->pg == iter2->pg)                ) {
```

resulting in:

```
[   68s] json_object_iterator.c: In function 'fjson_object_iter_equal':
[   68s] json_object_iterator.c:176:20: error: self-comparison always evaluates to true [-Werror=tautological-compare]
[   68s]          (iter2->pg == iter2->pg)                ) {
[   68s]                     ^~

```

Did you mean...

```
			if ( (iter1->curr_idx == iter2->curr_idx) &&
			     (iter1->pg == iter2->pg)                ) {
```

--- a/json_object_iterator.c
+++ b/json_object_iterator.c
@@ -173,7 +173,7 @@ fjson_object_iter_equal(const struct fjson_object_iterator* iter1,
 			is_eq = 1;
 		} else {
 			if ( (iter1->curr_idx == iter2->curr_idx) &&
-			     (iter2->pg == iter2->pg)                ) {
+			     (iter1->pg == iter2->pg)                ) {
 				is_eq = 1;
 			} else {
 				is_eq = 0;

